ansible/roles/test/tasks/ptf_runner_helper.yml

```diff
@@ -0,0 +1,104 @@
+        - ports_file=\"/tmp/ports.json\"
+        - dut_mac='{{ dut_mac }}'
+        - dut_vlan_ip='192.168.0.1'
+        - default_ip_range='192.168.0.0/16'
+        - vlan_ip_range='{{ vlan_ip_range }}'
+        - lo_v6_prefix='{{ lo_v6_prefix }}'
+        - arista_vms=\"['{{ vm_hosts | list | join("','") }}']\"
+        - preboot_files='{{ preboot_files }}'
+        - preboot_type='{{ item }}'
+
+  always:
+
+    - name: Set all the filename vars when there is no preboot type
+      set_fact:
+        reboot_log: '/tmp/{{reboot_type}}.log'
+        capture_pcap: '/tmp/capture.pcap'
+        filter_pcap: '/tmp/capture_filtered.pcap'
+        syslog_file: '/tmp/syslog'
+        sairedis_rec: '/tmp/sairedis.rec'
+        swss_rec: '/tmp/swss.rec'
+      when: not item or item == 'None'
+
+    - name: Set all the filename vars when there is a preboot type
+      set_fact:
+        reboot_log: '/tmp/{{reboot_type}}-{{item}}.log'
+        capture_pcap: '/tmp/capture_{{item}}.pcap'
+        filter_pcap: '/tmp/capture_filtered_{{item}}.pcap'
+        syslog_file: '/tmp/syslog_{{item}}'
+        sairedis_rec: '/tmp/sairedis.rec.{{item}}'
+        swss_rec: '/tmp/swss.rec.{{item}}'
+      when: item and item != 'None'
+
+    - name: Copy test results from ptf to the local box /tmp/*-reboot*.log
+      fetch: src="{{ reboot_log }}" dest='/tmp/' flat=true fail_on_missing=false
+      delegate_to: "{{ ptf_host }}"
+
+    - name: Copy pcap files from ptf to the local box /tmp/
+      fetch: src={{ item }} dest='/tmp/' flat=true fail_on_missing=false
+      delegate_to: "{{ ptf_host }}"
+      with_items:
+        - "{{ capture_pcap }}"
+        - "{{ filter_pcap }}"
+
+    - name: Extract all syslog entries since the last reboot
+      extract_log:
+        directory: '/var/log'
+        file_prefix: 'syslog'
+        start_string: 'Linux version'
+        target_filename: "{{ syslog_file }}"
+      become: yes
+
+    - name: Copy the exctracted syslog entries to the local machine
+      fetch:
+        src: "{{ syslog_file }}"
+        dest: '/tmp/'
+        flat: yes
+
+    - name: Extract all sairedis.rec entries since the last reboot
+      extract_log:
+        directory: '/var/log/swss'
+        file_prefix: 'sairedis.rec'
+        start_string: 'recording on:'
+        target_filename: "{{ sairedis_rec }}"
+
+    - name: Copy the exctracted sairedis.rec entries to the local machine
+      fetch:
+        src: "{{ sairedis_rec }}"
+        dest: '/tmp/'
+        flat: yes
+
+    - name: Extract all swss.rec entries since the last reboot
+      extract_log:
+        directory: '/var/log/swss'
+        file_prefix: 'swss.rec'
+        start_string: 'recording started'
+        target_filename: "{{ swss_rec }}"
+
+    - name: Copy the exctracted swss.rec entries to the local machine
+      fetch:
+        src: "{{ swss_rec }}"
+        dest: '/tmp/'
+        flat: yes
+
+
+- name: Wait for the DUT to be ready for the next test
+  pause: seconds=420
```

Comment:
>420 [](start = 17, length = 3)

Why pause so long?

Reply:
Ying suggested that during his tests he had seen that 7 mins is a good wait time for the system to be in a stable state before subsequent reboots